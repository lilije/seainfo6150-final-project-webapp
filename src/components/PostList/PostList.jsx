import React from 'react';
import styles from './PostList.module.css';

class PostList extends React.Component {
    
    displayPosts = () => {
        const listItems = this.props.posts.map((post, index) =>
        <a className={styles.post}  href={`/post/${post.id}`} key={index}>
            <img src={post.data.imageUrl} alt="post"/>
            <div className={styles.bottom}>
                <h2>{post.data.title}</h2>
                <div>
                    <span className={styles.author}>By {post.data.author}</span>
                    <span className={styles.price}>${post.data.price}</span>
                </div>
            </div>
        </a>
        );
        return (
            <div className={styles.root}>
                <div className={styles.topContent}>
                    <div className={styles.innerContainer}>
                        <h2 className={styles.title}>SIMPLIFY DINNER</h2>
                        <p className={styles.subtitle}>Fresh weekly meal kits customized to your liking and starting at $6.99 per serving.</p>
                    </div>
                    <div className={styles.secondRow}>
                        <div className={styles.secondRowContent}>
                            <div>
                                <img className={styles.secondRowContentImage} src="https://www.flaticon.com/svg/static/icons/svg/3830/3830217.svg" alt="post"/>
                            </div>
                            <div className={styles.secondRowContentTitle}>Select</div>
                            <div>Carb conscious, calorie conscious, vegetarian – we’ve got lots of options to fit your schedule and diet.</div>
                        </div>
                        <div className={styles.secondRowContent}>
                            <div>
                                <img className={styles.secondRowContentImage} src="https://www.flaticon.com/svg/static/icons/svg/1300/1300455.svg" alt="post" />
                            </div>
                            <div className={styles.secondRowContentTitle}>Customize It</div>
                            <div>Use our Customize It feature to upgrade, swap, or double up proteins on select meals.</div>
                        </div>
                        <div className={styles.secondRowContent}>
                            <div>
                                <img className={styles.secondRowContentImage} src="https://www.flaticon.com/premium-icon/icons/svg/2454/2454273.svg" alt="post"/>
                            </div>
                            <div className={styles.secondRowContentTitle}>Enjoy</div>
                            <div>Effortlessly create and plate your meals in no time. Dinner is solved!</div>
                        </div>
                    </div>
                </div>
                <div className={styles.thirdRow}>
                    <div className={styles.thirdRowContent}>
                        <div className={styles.thirdRowContentTitle}>More Than a Meal Kit</div>
                        <div className={styles.thirdRowContentSubTitle}>Weekly recipe rotations for all skill levels and dietary preferences means there's always something new and exciting to cook.</div>
                    </div>
                </div>
               {listItems}
            </div>
        );
    };

    render() {
        return (  <div className={styles.top}>{this.displayPosts()}</div> );
    }
}

export default PostList;
