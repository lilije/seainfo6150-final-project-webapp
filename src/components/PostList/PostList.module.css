.root {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 1500px;
  padding-bottom: 80px;
}

.topContent {
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: center;
  text-align: center;
  margin: 0 auto;
  margin-bottom: 100px;
}

.title {
  color: #4AE6CE;
  font-weight: 600;
}

.subtitle {
  color:  #293963;
  font-size: 24px;
}

.post {
  width: 100%;
  border-radius: 30px;
  display: flex;
  flex-direction: column;
  text-decoration: none;
  margin-bottom: 30px;
  box-shadow: 0 6px 10px rgba(0, 0, 0, 0.2);
  cursor: pointer;
}

.top {
  margin-top: 100px;
}

.price {
  color: #FF9800;
  font-weight: 600;
  margin-left: 60px;
}

h2 { color: #293963; }


img {
  width: 100%;
  max-width: 100%;
  border-top-left-radius: 30px;
  border-top-right-radius: 30px;
}

.author {
  color: #4AE6CE;
  font-weight: 600;
}

.bottom {
  display: flex;
  flex-direction: column;
  padding: 0 30px 30px;
  justify-content: space-between;
  height: 100%;
}

.secondRow {
  display: flex;
  margin-top: 80px;
  flex-direction: column;
  font-family: 'Courier New', Courier, monospace;
}

.secondRowContent {
  padding: 20px;
}

.secondRowContentImage {
  width: 80px;
}

.secondRowContentTitle {
  margin: 20px 0;
}

.thirdRow {
  height: 500px;
  width: 100%;
  display: flex;
  align-items: center;
  margin-bottom: 80px;
  text-align: center;
  justify-content: center;
  background-repeat:no-repeat;
  -webkit-background-size:cover;
  -moz-background-size:cover;
  -o-background-size:cover;
  background-size:cover;
  background-position:center;
  background-image: url(https://images.unsplash.com/photo-1576867757603-05b134ebc379?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=1950&q=80);
}

.thirdRowContent {
  height: 500px;
  width: 100%;
  flex-direction: column;
  background-color: #fff;
  opacity: .6;
}

.thirdRowContentTitle {
  font-size: 60px;
  color: #008600;
  padding: 20px 0;
}

.thirdRowContentSubTitle {
  font-size: 24px;
}

@media screen and (min-width: 800px) {
  .post {
    width: 48%;
  }
}

@media screen and (min-width: 1200px) {
  .secondRow {
    flex-direction: row;
  }
  
  .innerContainer {
    width: 40%;
  }
  
  .secondRowContent {
    width: 33%;
  }
  
  .thirdRowContent {
    height: 200px;
    width: 60%;
    flex-direction: column;
    background-color: #fff;
    opacity: .6;
  }
  
  .post {
    width: 32%;
  }
}