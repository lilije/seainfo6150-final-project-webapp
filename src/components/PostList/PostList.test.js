import React from "react";
import PostList from "./PostList";
import { render } from "@testing-library/react";

const posts = [
  {
    data : {
      id: 'sdafsdfsdf',
      imageUrl: 'https://homechef.imgix.net/https%3A%2F%2Fasset.homechef.com%2Fuploads%2Fmeal%2Fplated%2F11888%2F419362.002.01SpanishPorkTacos_Ecomm_copy.jpg?ixlib=rails-1.1.0&w=850&auto=format&s=eadb858afc60591a00b0df808ee2a2b3',
      price: 16,
      category: 'taco',
      author: 'lili',
      description: 'Weve heard of smoke getting in your eyes, or smoke in the water, but smoky tacos? Now weve heard everything! But why not? A hint of paprika (the smoky!) on the pork, a bit of jalapeÃ±o ranch from the slaw, and fresh pico combine for handheld dinner delight. This meal is smokin'
    }
  }
]

describe("PostList tests", () => {
  it("renders correctly", () => {
    const { container } = render(<PostList posts={posts} />);
    expect(container).toMatchSnapshot();
  });
});
