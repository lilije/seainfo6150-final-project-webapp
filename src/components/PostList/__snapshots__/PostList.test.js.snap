// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList tests renders correctly 1`] = `
<div>
  <div
    class="top"
  >
    <div
      class="root"
    >
      <div
        class="topContent"
      >
        <div
          class="innerContainer"
        >
          <h2
            class="title"
          >
            SIMPLIFY DINNER
          </h2>
          <p
            class="subtitle"
          >
            Fresh weekly meal kits customized to your liking and starting at $6.99 per serving.
          </p>
        </div>
        <div
          class="secondRow"
        >
          <div
            class="secondRowContent"
          >
            <div>
              <img
                alt="post"
                class="secondRowContentImage"
                src="https://www.flaticon.com/svg/static/icons/svg/3830/3830217.svg"
              />
            </div>
            <div
              class="secondRowContentTitle"
            >
              Select
            </div>
            <div>
              Carb conscious, calorie conscious, vegetarian – we’ve got lots of options to fit your schedule and diet.
            </div>
          </div>
          <div
            class="secondRowContent"
          >
            <div>
              <img
                alt="post"
                class="secondRowContentImage"
                src="https://www.flaticon.com/svg/static/icons/svg/1300/1300455.svg"
              />
            </div>
            <div
              class="secondRowContentTitle"
            >
              Customize It
            </div>
            <div>
              Use our Customize It feature to upgrade, swap, or double up proteins on select meals.
            </div>
          </div>
          <div
            class="secondRowContent"
          >
            <div>
              <img
                alt="post"
                class="secondRowContentImage"
                src="https://www.flaticon.com/premium-icon/icons/svg/2454/2454273.svg"
              />
            </div>
            <div
              class="secondRowContentTitle"
            >
              Enjoy
            </div>
            <div>
              Effortlessly create and plate your meals in no time. Dinner is solved!
            </div>
          </div>
        </div>
      </div>
      <div
        class="thirdRow"
      >
        <div
          class="thirdRowContent"
        >
          <div
            class="thirdRowContentTitle"
          >
            More Than a Meal Kit
          </div>
          <div
            class="thirdRowContentSubTitle"
          >
            Weekly recipe rotations for all skill levels and dietary preferences means there's always something new and exciting to cook.
          </div>
        </div>
      </div>
      <a
        class="post"
        href="/post/undefined"
      >
        <img
          alt="post"
          src="https://homechef.imgix.net/https%3A%2F%2Fasset.homechef.com%2Fuploads%2Fmeal%2Fplated%2F11888%2F419362.002.01SpanishPorkTacos_Ecomm_copy.jpg?ixlib=rails-1.1.0&w=850&auto=format&s=eadb858afc60591a00b0df808ee2a2b3"
        />
        <div
          class="bottom"
        >
          <h2 />
          <div>
            <span
              class="author"
            >
              By 
              lili
            </span>
            <span
              class="price"
            >
              $
              16
            </span>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
